packages/ag-charts-enterprise/src/features/zoom/zoom.ts

```diff
@@ -288,11 +291,30 @@ export class Zoom extends _ModuleSupport.BaseModuleInstance implements _ModuleSu
         if (!this.enabled) return;
 
         const {
-            series: { paddedRect, shouldFlipXY },
+            series: { paddedRect, shouldFlipXY, minRect },
         } = event;
 
         this.seriesRect = paddedRect;
         this.shouldFlipXY = shouldFlipXY;
+
+        if (!paddedRect || !minRect) return;
+
+        const zoom = definedZoomState(this.zoomManager.getZoom());
+
+        if (this.isScalingX()) {
+            const widthRatio = (minRect.width * this.minVisibleItemsX) / paddedRect.width;
+            const normalisedWidthRatio = widthRatio * (zoom.x.max - zoom.x.min);
+            this.minRatioX ||= Math.min(1, round(normalisedWidthRatio, DECIMALS));
+        }
+
+        if (this.isScalingY()) {
+            const heightRatio = (minRect.height * this.minVisibleItemsY) / paddedRect.height;
+            const normalisedHeightRatio = heightRatio * (zoom.y.max - zoom.y.min);
+            this.minRatioY ||= Math.min(1, round(normalisedHeightRatio, DECIMALS));
+        }
+
+        this.minRatioX ||= this.minRatioY || 0;
```

Comment:
When the chart first loads, the `getMinRect()` looks at the `contextNodeData` but it is an empty array initially. So the width and height of the rect are 0.

When the next layout completes, i.e. next interaction, it looks again and calculates it correctly.

Using `||=` instead of `??=` ensures it gets set to something after the first 0 but also prevents it from fluctuating due to fractions as you zoom in, causing inconsistencies in the zoom limits.

Perhaps instead it'd be better if the layout event returns `undefined` if there's no `contextNodeData`.

I've just realised this should check the series rect as well, as it'll need to recalculate when that changes.

Reply:
I've updated this to use `undefined` for empty `contextNodeData`, but it remains simpler to use `||=` for these ratios and default them to 0.